### PR TITLE
fix: add tensor parallel size check

### PR DESCRIPTION
**Pull request by aiwantaozi**
#814

### CODE REVIEW

File: gpustack/policies/candidate_selectors/vllm_resource_fit_selector.py [hunk 1]
```diff
@@ -70,11 +70,30 @@ async def estimate_model_vram(model: Model) -> int:
     return weight_size * 1.2 + cuda_graph_size
 
 
+def get_model_num_attention_heads(model: Model) -> int:
```
Comment (gitlawr) [inline gpustack/policies/candidate_selectors/vllm_resource_fit_selector.py]: -> Optional[int]

File: gpustack/policies/candidate_selectors/vllm_resource_fit_selector.py [hunk 2]
```diff
@@ -70,11 +70,30 @@ async def estimate_model_vram(model: Model) -> int:
...
+    """
+    Get the number of attention heads in the model.
+    """
+
+    num_attention_heads = None
+    try:
+        config = get_pretrained_config(model, trust_remote_code=True)
```
Comment (gitlawr) [inline gpustack/policies/candidate_selectors/vllm_resource_fit_selector.py:80]: Why true v.s. getting from the parameters?

Comment (aiwantaozi) [reply gpustack/policies/candidate_selectors/vllm_resource_fit_selector.py:80]: without setting parameter will fail to get the config

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**